src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_REG_DATA_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_REG_DATA_LENGTH) {
+            throw new RegistrationException("Login must be at least 6 characters long");
+        }
+        for (int i = 0; i < Storage.people.size(); i++) {
+            if (Storage.people.get(i).getLogin().equals(user.getLogin())) {
+                throw new RegistrationException("This login is already in use");
+            }
+        }
```

Comment:
Implement method, e.g. Storage.get(Login) 

and use it to reduce code here
```suggestion
        for (int i = 0; i < Storage.people.size(); i++) {
            if (Storage.people.get(i).getLogin().equals(user.getLogin())) {
                throw new RegistrationException("This login is already in use");
            }
        }
```